docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,356 @@
+
+<br/>
+
+## Update the folder structure automatically
+
+:::prerequisites
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/codemod-prerequisites.md)!!!
+:::
+
+::: caution
+The codemod creates a new Strapi v4 plugin, leaving the Strapi v3 plugin in place. We recommend confirming the v4 version of the plugin is working properly before deleting the v3 version.
+:::
+
+The [`update-plugin-folder-structure` codemod](https://github.com/strapi/codemods/blob/main/migration-helpers/update-plugin-folder-structure.js) can be used to automatically update the folder structure of a plugin for Strapi v4. To execute the codemod, run the following commands in a terminal:
+
+```sh
+cd <the-folder-where-the-strapi-codemods-repo-was-cloned>
+node ./migration-helpers/update-plugin-folder-structure.js <path-to-v3-plugin> <path-for-v4-plugin>
+```
+
+
+The codemod:
+
+- creates 2 entry files: `strapi-server.js` and `strapi-admin.js`,
+- organizes files and folders into a `server` and an `admin` folders, respectively,
+- converts `models` to `contentTypes`,
+- and exports `services` as functions.
+
+## Update the folder structure manually
+
+Manually updating the folder structure requires moving and updating the content of multiple files and folders. These steps are described in the following subsections.
+
+:::note
+The folder structure is given as an example, and files and folders can be organized freely as long as `strapi-server.js` and `strapi-admin.js` exist and import all the required files.
+:::
+
+### Create a `server` folder
+
+The `server` folder includes all the code for the back end of the plugin. To create it at the root of the plugin folder, run the following command in a terminal:
+
+```sh
+cd <my-plugin-folder-name>
+mkdir server
+```
+
+### Move controllers, services, and middlewares
+
+:::callout
+In Strapi v3, controllers, services, and middlewares of a plugin should follow a strict folder structure convention.
+
+In Strapi v4, the organization of files and folders for plugins is flexible. However, it is recommended to create dedicated folders for [controllers](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#controllers), [services](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#services), and [middlewares](/developer-docs/latest/developer-resources/plugin-api-reference/server.html#middlewares) inside a `server` folder.
+
+:::
+
+To update the controllers, services, and middlewares of a plugin to Strapi v4, create specific subfolders in a `server` folder.
+
+Plugin files and folders in Strapi v4 should meet 2 requirements:
+
+- Each file in the `server/<subfolder-name>` should export a function taking the `strapi` instance (object) as a parameter and returning an object.
+- Each of the `server/<subfolder-name>` folders should include an `index.js` file that exports all files in the folder.
```

Comment:
Are the paths `server/<subfolder-name>` correct?

Reply:
Not sure to get what you mean, @petersg83 ? What else should we name them?